capsules/src/process_console.rs

```diff
@@ -964,9 +1067,75 @@ impl<'a, A: Alarm<'a>, C: ProcessManagementCapability> uart::ReceiveClient
                                 // Backspace, echo and remove last byte
                                 // Note echo is '\b \b' to erase
                                 let _ = self.write_bytes(&['\x08' as u8, ' ' as u8, '\x08' as u8]);
-                                command[index - 1] = '\0' as u8;
+                                command[index - 1] = TERMINATOR;
                                 self.command_index.set(index - 1);
                             }
+                        } else if read_buf[0] == ESC || self.control_seq_in_progress.get() {
```

Comment:
Guard this with the same `if` that allows the compiler to delete it if the commands list length is 0.

Reply:
Will the compiler delete the code if I write `COMMAND_HISTORY_LEN > 0 && (read_buf[0] ...` or should the `COMMAND_HISTORY_LEN >0` be outside the other conditions?

Reply:
A good compiler should do that, as the first condition always evaluates to false. You can write it like that, compile the code (using 0) and look at the code size. Do the same thing again while splitting into two statements and look at the code size.